lisa/analysis/rta.py

```diff
@@ -229,11 +301,83 @@ def df_rtapp_stats(self, task=None):
         df = self._get_stats()
         return self._task_filtered(df, task)
 
+    @memoized
+    @df_rtapp_stats.used_events
+    def df_phases(self, task):
+        """
+        Get phases actual start times and durations
+
+        :param task: Name of the task that we want the phases dataframe of
+        :type task: def __str__(self):
+
+        :returns: A :class:`pandas.DataFrame` with index representing the
+        start time of a phase and these column:
+
+        * ``duration``: the measured phase duration.
+        """
+        # Mark for removal all the events that are not the first 'start'
+        def keep_first_start(raw):
+            if raw.phase_loop:
+                return -1
+            if raw.event == 'end':
+                return -1
+            return 0
+
+        loops_df = self.df_rtapp_loop(task)
+
+        # Keep only the first 'start' and the last 'end' events
+        # Do that by first setting -1 the 'phase_loop' of all entries which are
+        # not the first 'start' event. Then drop the 'event' column so that we
+        # can drop all duplicates thus keeping only the last 'end' even for
+        # each phase.
+        phases_df = loops_df[['event', 'phase', 'phase_loop']].copy()
+        phases_df['phase_loop'] = phases_df.apply(keep_first_start, axis=1)
+        phases_df = phases_df[['phase', 'phase_loop']]
+        phases_df.drop_duplicates(keep='last', inplace=True)
+
+        # Compute deltas and keep only [start..end] intervals
+        durations = list(phases_df.index[1:] - phases_df.index[:-1])
+        durations = durations[::2]
```

Comment:
Why keep only one item out of two in the list ?

Reply:
in L339 we compute all the deltas:
```
[
 (phase1_start, phase_1end),
 (phase1_end,   phase2_start),
 (phase2_start, phase2_end),...
]
```

but we actually need only the `(*_start, *_end)` intervals.

Will add a note on the inline comment.